mcstatus/motd.py

```diff
@@ -0,0 +1,558 @@
+    MinecraftColor.DARK_PURPLE: ("#AA00AA", "#2A002A"),
+    MinecraftColor.GOLD: ("#FFAA00", "#402A00"),
+    MinecraftColor.GRAY: ("#AAAAAA", "#2A2A2A"),
+    MinecraftColor.DARK_GRAY: ("#555555", "#151515"),
+    MinecraftColor.BLUE: ("#5555FF", "#15153F"),
+    MinecraftColor.GREEN: ("#55FF55", "#153F15"),
+    MinecraftColor.AQUA: ("#55FFFF", "#153F3F"),
+    MinecraftColor.RED: ("#FF5555", "#3F1515"),
+    MinecraftColor.LIGHT_PURPLE: ("#FF55FF", "#3F153F"),
+    MinecraftColor.YELLOW: ("#FFFF55", "#3F3F15"),
+    MinecraftColor.WHITE: ("#FFFFFF", "#3F3F3F"),
+    MinecraftColor.MINECOIN_GOLD: ("#DDD605", "#373501"),
+}
+
+MINECRAFT_COLOR_TO_HEX_JAVA = MINECRAFT_COLOR_TO_HEX_BEDROCK.copy()
+MINECRAFT_COLOR_TO_HEX_JAVA[MinecraftColor.GOLD] = ("#FFAA00", "#2A2A00")
+
+MINECRAFT_COLOR_TO_ANSI = {
+    MinecraftColor.BLACK: "30",
+    MinecraftColor.DARK_BLUE: "34",
+    MinecraftColor.DARK_GREEN: "32",
+    MinecraftColor.DARK_AQUA: "36",
+    MinecraftColor.DARK_RED: "31",
+    MinecraftColor.DARK_PURPLE: "35",
+    MinecraftColor.GOLD: "33",
+    MinecraftColor.GRAY: "37",
+    MinecraftColor.DARK_GRAY: "90",
+    MinecraftColor.BLUE: "94",
+    MinecraftColor.GREEN: "92",
+    MinecraftColor.AQUA: "96",
+    MinecraftColor.RED: "91",
+    MinecraftColor.LIGHT_PURPLE: "95",
+    MinecraftColor.YELLOW: "93",
+    MinecraftColor.WHITE: "97",
+    MinecraftColor.MINECOIN_GOLD: "33",
+}
+
+
+@dataclass
+class WebColor:
+    """Raw HTML color from MOTD.
+
+    Can be found in MOTD when someone uses gradient.
+    """
+
+    hex: str
+    rgb: tuple[int, int, int]
+
+    @classmethod
+    def from_hex(cls, hex: str) -> Self:
+        """Parse hex color, from RGB.
+
+        :raises ValueError: When hex color isn't correct, so we can't generate RGB for it.
+        :returns: New ``WebColor`` instance.
+        """
+        hex = hex.lstrip("#")
+
+        try:
+            rgb = tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))
+        except ValueError:
+            raise ValueError(f"Failed to parse given hex color: {'#' + hex!r}")
+
+        return cls.from_rgb(cast("tuple[int, int, int]", rgb))
+
+    @classmethod
+    def from_rgb(cls, rgb: tuple[int, int, int]) -> Self:
+        """Parse RGB color, from hex.
+
+        :raises ValueError: When RGB color is out of its 8-bit range.
+        :returns: New ``WebColor`` instance.
+        """
+        if not all(255 >= e >= 0 for e in rgb):
+            raise ValueError("Got RGB color out of its 8-bit range (0-255)")
+
+        hex = "#{:02x}{:02x}{:02x}".format(*rgb)
+        return cls(hex, rgb)
+
+
+@dataclass
+class TranslationTag:
+    """Represents a ``translate`` field in server's answer.
+
+    This just exist, and completely ignored by our end-parsers (``ansi``, ``html`` etc. properties).
+    You can parse them with ``Motd.parsed`` attribute.
+
+    .. seealso:: `Minecraft's wiki. <https://minecraft.fandom.com/wiki/Raw_JSON_text_format#Translated_Text>`_
```

Comment:
Why the underscore at the end?

Reply:
Because it's RST syntax.

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#external-links